Sources/GRPCHTTP2TransportNIOPosix/TLSConfig.swift

```diff
@@ -217,3 +217,54 @@ extension HTTP2ServerTransport.Posix.Config {
     }
   }
 }
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+extension HTTP2ClientTransport.Posix.Config {
+  /// The security configuration for this connection.
+  public struct TransportSecurity: Sendable {
+    package enum Wrapped: Sendable {
+      case plaintext
+      case tls(TLS)
+    }
+
+    package let wrapped: Wrapped
+
+    /// This connection is plaintext: no encryption will take place.
+    public static let plaintext = Self(wrapped: .plaintext)
+
+    /// This connection will use TLS.
+    public static func tls(_ tls: TLS) -> Self {
+      Self(wrapped: .tls(tls))
+    }
+  }
+
+  public struct TLS: Sendable {
+    /// The certificates the client will offer during negotiation.
+    public var certificateChain: [TLSConfig.CertificateSource]
+
+    /// The private key associated with the leaf certificate.
+    public var privateKey: TLSConfig.PrivateKeySource?
+
+    /// How to verify the server certificate, if one is presented.
+    public var serverCertificateVerification: TLSConfig.CertificateVerification
+
+    /// The trust roots to be used when verifying server certificates.
+    public var trustRoots: TLSConfig.TrustRootsSource
+
+    /// Create a new HTTP2 NIO Posix transport TLS config, with some values defaulted:
+    /// - `certificateChain` is `[]`
+    /// - `privateKey` is `nil`
+    /// - `serverCertificateVerification` equals `fullVerification`
+    /// - `trustRoots` equals `systemDefault`
+    ///
+    /// - Returns: A new HTTP2 NIO Posix transport TLS config.
+    public static func defaults() -> Self {
+      Self.init(
+        certificateChain: [],
+        privateKey: nil,
+        serverCertificateVerification: .fullVerification,
+        trustRoots: .systemDefault
+      )
+    }
```

Comment:
Can we add an mTLS helper too?

Reply:
Ah, I didn't because I don't think anything changes? I may be wrong here but the client will verify server certificates in either "normal" TLS or mTLS, so the configuration is the same in both cases.

Reply:
In mTLS the client will present its own cert to the server as well. We have some tests in v1 which cover this: https://github.com/grpc/grpc-swift/blob/6f396ca3aed37ecb4f9c7b4d6fbb0b98df883860/Tests/GRPCTests/MutualTLSTests.swift#L156